.github/workflows/integration-tests-release.yaml

```diff
@@ -0,0 +1,219 @@
+          fi
+
+      - name: Run Integration Tests
+        if: ${{ env.IT_CAN_RUN == 'yes' }}
+        env:
+          SCENARIO: ${{ matrix.scenario }}
+          K8SSANDRA_CASSANDRA_VERSION: ${{ matrix.cassandra_version }}
+        run: |
+          # Set a valid name for the artifacts in case of failure
+          echo "ARTIFACT_NAME=k8s-artifacts-${K8SSANDRA_CASSANDRA_VERSION}-$(echo $SCENARIO|tr '":/' '___')" >> $GITHUB_ENV
+          make kind-integ-test TESTS="$SCENARIO" GO_FLAGS="-v" CLUSTER_CLEANUP="success"
+
+      - name: Collect artifacts
+        if: ${{ failure() }}
+        env:
+          ARTIFACTS_DIR: artifacts
+        run: tests/integration/scripts/collect_artifacts.sh
+
+      - uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: k8s-artifacts-${{ env.ARTIFACT_NAME }}
+          path: artifacts/
+      
+      # Uncomment to run with SSH on failure
+      - name: Setup tmate session
+        if: "failure() && contains(github.event.head_commit.message, '[SSH]')"
+        uses: mxschmitt/action-tmate@v3
+        timeout-minutes: 10
+  
+  release:
+    if: |
+      ${{ (( github.event_name == 'release' && github.event.action == 'published' ) 
+        || ( github.event_name == 'push' && github.ref == 'refs/heads/main' )) }}
+    runs-on: ubuntu-latest
+    needs: integration-tests
+    env:
+      HELM_VERSION: 3.5.3
+      YQ_VERSION: v4.6.3
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Update PATH
+        run: |
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+
+      - name: Install and validate tool versions
+        run: |
+          scripts/install-helm.sh $HELM_VERSION
+          scripts/install-yq.sh $YQ_VERSION
+          helm version
+          yq -V
+
+      - name: Update chart dependencies
+        run: |
+          scripts/update-helm-deps.sh
+
+      - name: Compute release chart version
+        id: compute_next_version
+        run: |
+          DATE_TIME=$(date '+%Y%m%d%H%M%S')
+          RELEASE_VERSION=$(echo $(yq e '.version' charts/k8ssandra/Chart.yaml) | sed "s/-.*/-${DATE_TIME}-${GITHUB_SHA::8}/")
+          echo "Release version is: $RELEASE_VERSION"
+          echo "RELEASE_VERSION=${RELEASE_VERSION}" >> $GITHUB_ENV
+          echo "::set-output name=RELEASE_VERSION::${RELEASE_VERSION}"
+
+      - name: Update Helm chart version
+        run: |
+          yq eval ".version |= \"${RELEASE_VERSION}\"" charts/k8ssandra/Chart.yaml -i
+          cat charts/k8ssandra/Chart.yaml
+
+      - name: Create working directory and copy charts
+        run: |
+          mkdir build
+          cp -rv charts/* build/
+
+      - name: Generate package archives
+        run: |
+          ls | while read c; do
+            if [[ -d $c ]]; then
+              helm package $c
+            fi
+          done
+        working-directory: build
+
+      - name: Check if Helm release is possible
```

Comment:
Shouldn't this just be in the original if clause? That the whole job is skipped if this isn't true?

Reply:
I'm not sure I can put this in an `if:` clause as is actually. I'll check how I can make it so that the whole job is skipped, it's cleaner indeed.

Reply:
I made it so that the release job only runs if we're not on a fork.